Container vulnerability collector for gcr.io registry (#1250)
kyma-project/test-infra: development/tools/cmd/vulnerabilitycollector/README.md

# Vulnerability Collector

## Overview

> The tool queries the public google container repository API for all containers it can find for a given project, while traversing paths as they're found. With the generated list of containers, it will pick the last built container and runs a check for any security vulnerabilities found from the GCR.io vulnerability scanner. If a region or path is specified, the number of queries is drastically reduced.

Reviewer: The overview does not really say what the Vulnerability Collector is and what it does. Please rewrite the section. Is its only function to query the public Google container repository API?

Author: As of now, this is the only thing we do with it, yes.

Reviewer: ```suggestion
The Vulnerability Collector queries the public Google container repository API about all containers in a given project while traversing paths as they're found. The result is a generated list of containers. The Vulnerability Collector picks the latest container and runs a check for any security vulnerabilities found by the GCR.io vulnerability scanner. If a region or path is specified, the number of queries is drastically reduced.
```

If the region parameter is left out, the global region is assumed. If it is passed, the region is checked if it is either `eu` or `us`.

## Usage

> To run it, use:

Reviewer: ```suggestion
To use the Vulnerability Collector, run:
```

```bash
go run main.go -project="kyma-project" -region="eu" -path="test-infra"
```

> ## Output

Reviewer: ```suggestion
### Output

See the following output example provided by the Vulnerability Collector.
```

# Vulnerability Collector

## Overview
The tool queries the public google container repository API for all containers it can find for a given project, while traversing paths as they're found. With the generated list of containers, it will pick the last built container and runs a check for any security vulnerabilities found from the GCR.io vulnerability scanner. If a region or path is specified, the number of queries is drastically reduced.

> If the region parameter is left out, the global region is assumed. If it is passed, the region is checked if it is either `eu` or `us`.

Reviewer: ```suggestion
If the **region** parameter is left out, the **global region** is taken into account. If the **region** parameter is provided, it is checked if it is either `eu` or `us`.
```
